src/main/java/com/hazelcast/jdbc/HazelcastJdbcPreparedStatement.java

```
@@ -0,0 +1,370 @@
+        checkClosed();
+        throw JdbcUtils.unsupported("Binary Stream is not supported");
+    }
+
+    @Override
+    public void setCharacterStream(int parameterIndex, Reader reader, long length) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("Character Stream is not supported");
+    }
+
+    @Override
+    public void setAsciiStream(int parameterIndex, InputStream x) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("Ascii Stream is not supported");
+    }
+
+    @Override
+    public void setBinaryStream(int parameterIndex, InputStream x) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("Binary Stream is not supported");
+    }
+
+    @Override
+    public void setCharacterStream(int parameterIndex, Reader reader) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("Character Stream is not supported");
+    }
+
+    @Override
+    public void setNCharacterStream(int parameterIndex, Reader value) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("NCharacter Stream is not supported");
+    }
+
+    @Override
+    public void setClob(int parameterIndex, Reader reader) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("Clob is not supported");
+    }
+
+    @Override
+    public void setBlob(int parameterIndex, InputStream inputStream) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("Blob is not supported");
+    }
+
+    @Override
+    public void setNClob(int parameterIndex, Reader reader) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("NClob is not supported");
+    }
+
+    private void setParameter(int parameterIndex, Object parameter) {
+        parameters.add(parameterIndex - 1, parameter);
```

The implementation of this method is incorrect - it will throw an exception in many situations because we do not ensure that the list has sufficient capacity. We should check the list size first, and if it is not sufficient - expand it. Also, this indicates that there are not enough tests for parameters.

Another problem is that there is no check for parameter index (it should be positive).

Last, how to deal with an incomplete parameter list? Suppose that a user has a query `SELECT ... WHERE ? = ?`. There are two parameters. What should happen if the user did `setParameter(2, ...)`, but didn't do `setParameter(1, ...)`. Perhaps, we should throw an exception in this case.

---

For the parameters list, we are using `ArrayList` so the capacity check is redundant as it will expand automatically.
But for paramIndex and incomplete param list, the validation will add